### PR TITLE
[WIP] MatricesGTest.hpp & test_centrality_py: adds testing functions

### PR DESCRIPTION
**Pull request by PetholzA**
adds testMatrixToGraph() in cpp, and new test functions for: test_centrality test_clique and test_community in python 

this not the final commit! it is meant to to check if the coverage increases

### CODE REVIEW

**Comment by PetholzA** [comment]
note that the python test cases failed, because the previous PR  #1059 hasnt been pulled yet, if i change my files according to #1059 locally it works just fine

**Comment by fabratu** [comment]
@PetholzA Please rebase on top of master.

File: networkit/cpp/algebraic/test/MatricesGTest.cpp [hunk 1]
```diff
@@ -120,11 +120,26 @@ class MatricesGTest : public testing::Test {
...
 };
 
+template <class Matrix>
+void MatricesGTest::testMatrixToGraph() { 
+    
+    METISGraphReader graphReader;
+    Matrix mat = Matrix::adjacencyMatrix(graphReader.read("input/power.graph"));
+    Graph G =MatrixTools::matrixToGraph(mat);
+
+    ASSERT_EQ(G.numberOfEdges(),6594);
```
Comment (angriman) [inline networkit/cpp/algebraic/test/MatricesGTest.cpp:138]:
```suggestion
    EXPECT_EQ(G.numberOfEdges(), 6594);
```
See http://google.github.io/googletest/reference/assertions.html

File: networkit/cpp/algebraic/test/MatricesGTest.cpp [hunk 2]
```diff
@@ -1200,6 +1215,12 @@ TEST_F(MatricesGTest, testDimension) {
     testDimension<DenseMatrix>();
 }
 
+TEST_F(MatricesGTest, testMatrixToGraph) {
+    testMatrixToGraph<DynamicMatrix>();
+    testMatrixToGraph<CSRMatrix>();
+   // testMatrixToGraph<DenseMatrix>();
```
Comment (angriman) [inline networkit/cpp/algebraic/test/MatricesGTest.cpp:1221]: Why is this commented out?

File: networkit/cpp/algebraic/test/MatricesGTest.cpp [hunk 3]
```diff
@@ -1355,4 +1376,6 @@ TEST_F(MatricesGTest, testCSRMatrixSort) {
     CSRGeneralMatrix<double> csr(4, 4, triplets);
     csr.sort();
 }
+
+
```
Comment (angriman) [inline networkit/cpp/algebraic/test/MatricesGTest.cpp:1380]: Drop.

File: networkit/cpp/algebraic/test/MatricesGTest.cpp [hunk 4]
```diff
@@ -38,7 +38,7 @@ class MatricesGTest : public testing::Test {
 public:
     MatricesGTest() = default;
     virtual ~MatricesGTest() = default;
-
+
```
Comment (angriman) [inline networkit/cpp/algebraic/test/MatricesGTest.cpp:41]: Drop.

File: networkit/cpp/algebraic/test/MatricesGTest.cpp [hunk 5]
```diff
@@ -120,11 +120,26 @@ class MatricesGTest : public testing::Test {
...
     // TODO: Test other matrix classes
 
     // TODO: Test mmT multiplication, etc.!
 };
 
+template <class Matrix>
+void MatricesGTest::testMatrixToGraph() { 
+    
+    METISGraphReader graphReader;
+    Matrix mat = Matrix::adjacencyMatrix(graphReader.read("input/power.graph"));
```
Comment (angriman) [inline networkit/cpp/algebraic/test/MatricesGTest.cpp:135]:
```suggestion
    Matrix mat = Matrix::adjacencyMatrix(METISGraphReader{}.read("input/power.graph"));
```

File: networkit/test/test_centrality.py [hunk 6]
```diff
@@ -364,6 +382,5 @@ def testTopHarmonicCloseness(self):
 			for score_thc, score in zip(thc.topkScoresList(True), sorted(scores, reverse=True)):
 				self.assertAlmostEqual(score_thc, score, delta=tol)
 
-
```
Comment (angriman) [inline networkit/test/test_centrality.py:367]: Drop.

**Review by angriman — CHANGES_REQUESTED** [review]
Thanks for adding new tests. In addition to the comments I left, my concern is that this PR does too many things at once (adds tests for different modules). I recommend splitting this PR in 4 different PRs (or commits):
- Add C++ MatricesGTests
- Add Python matrices tests
- Add Python max clique tests
- Add Python community tests.

**Pull request closed without merging**